base/src/main/java/com/austindroids/austinfeedsme/common/events/EventsPresenter.kt

```diff
@@ -0,0 +1,146 @@
+        loadEvents(false)
+    }
+
+    override fun searchEvents(searchTerm: String) {
+        if (searchTerm.equals("reset", ignoreCase = true)) {
+            loadEvents()
+            return
+        }
+
+        val lowerCaseSearch = searchTerm.toLowerCase()
+
+        val subscribe = repository.eventsRX!!
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ events ->
+                    val filteredEvents = events.filter {
+                        !it.isFood
+                                || it.time < Date().time
+                                || TextUtils.isEmpty(it.description)
+                                || !it.description.toLowerCase().contains(lowerCaseSearch)
+                    }
+
+                    view.showEvents(filteredEvents)
+                }, { throwable -> Timber.e(Exception(throwable)) })
+    }
+
+    override fun setFiltering(eventsFilterType: EventsFilterType) {
+        currentFiltering = eventsFilterType
+    }
+
+    private fun filterEventsOnTimeSelection(events: List<Event>): ArrayList<Event> {
+        val aMinuteFromMidnight = DateUtils.aMinuteFromMinuteToday()
+        val sevenDaysFromNow = DateUtils.sevenDaysFromNow()
+
+        val eventsToShow = ArrayList<Event>()
+        for (event in events) {
+            when (currentFiltering) {
+                EventsFilterType.ALL_EVENTS -> eventsToShow.add(event)
+                EventsFilterType.TODAYS_EVENTS -> if (event.time < aMinuteFromMidnight) {
+                    eventsToShow.add(event)
+                }
+                EventsFilterType.THIS_WEEKS_EVENTS -> if (event.time < sevenDaysFromNow) {
+                    eventsToShow.add(event)
+                }
+                else -> eventsToShow.add(event)
+            }
+        }
+        return eventsToShow
+    }
+
+    private fun setEventCounts(eventsToShow: ArrayList<Event>) {
+        view.setTotalCount(eventsToShow.size)
+
+        val yummyCounts = HashMap<String, Int>()
+        for (event in eventsToShow) {
+
+            if (event.foodType == null || event.foodType == NONE.toString()) {
+                continue
+            }
+
+            if (event.foodType == PIZZA.name) {
+                val previousValue = yummyCounts[PIZZA.name]
+                yummyCounts[PIZZA.name] = if (previousValue == null) 1 else previousValue + 1
+            }
+            if (event.foodType == BEER.name) {
+                val previousValue = yummyCounts[BEER.name]
+                yummyCounts[BEER.name] = if (previousValue == null) 1 else previousValue + 1
+            }
+            if (event.foodType == TACO.name) {
+                val previousValue = yummyCounts[TACO.name]
+                yummyCounts[TACO.name] = if (previousValue == null) 1 else previousValue + 1
+            }
+
+        }
+
+        view.setPizzaCount(if (yummyCounts[PIZZA.name] == null) 0 else yummyCounts[PIZZA.name])
```

Comment:
use `?:` for a `null` handling instead of if/else, e.g. `setPizzaCount(yummyCounts[PIZZA.name] ?: 0)`